libraries/chain/block_state.cpp

```diff
@@ -48,6 +50,9 @@ block_state::block_state(const block_state_legacy& bsp) {
    active_proposer_policy->active_time = bsp.timestamp();
    active_proposer_policy->proposer_schedule = bsp.active_schedule;
    header_exts = bsp.header_exts;
+
+   auto& updated_if_extension = std::get<instant_finality_extension>(header_exts.lower_bound(if_ext_id)->second);
```

Comment:
Why is this needed, why not just use:
`const auto& if_extension = std::get<instant_finality_extension>(*ext);`

Reply:
This actually changes the new `header_exts` from the new `block_state` created in this constructor.
``header_exts`` is in the incoming `block_state_legacy` and I was not sure about mutating it.

Reply:
I could do it otherwise. I'd have to mutate `bsp.header_exts` instead of extracting a new map in `bsp.block->extract_header_extension(if_ext_id);`.

Reply:
It turns out that I originally added this in the wrong spot. It is now removed and we do set the `lib` in the qc_claim in the actual first transition block (see change in libraries/chain/block_header_state_legacy.cpp)